torch/nn/parallel/distributed.py

```diff
@@ -305,14 +305,16 @@ class DistributedDataParallel(Module):
 
     Args:
         module (Module): module to be parallelized
-        device_ids (list of int or torch.device): CUDA devices. This should
-                   only be provided when the input module resides on a single
-                   CUDA device. For single-device modules, the i'th
-                   :attr:`module` replica is placed on ``device_ids[i]``. For
-                   multi-device modules and CPU modules, ``device_ids`` must be
-                   ``None`` or an empty list, and input data for the forward
-                   pass must be placed on the correct device. (default: all
-                   visible devices for single-device modules)
+        device_ids (list of int or torch.device): CUDA devices.
+                   1) For single-device modules, ``device_ids`` should only
+                   contain exactly one device id, which represents the only
+                   CUDA device where the input module resides.
+                   Alternatively, ``device_ids`` can also be ``None``.
+                   2) For multi-device modules and CPU modules,
+                   ``device_ids`` must be ``None``.
+                   When ``device_ids`` is ``None`` for both cases,
+                   input data for the forward pass must be placed on the correct device.
+                   (default: all visible devices for single-device modules)
```

Comment:
is this default still relevant? Should we set default to None? Otherwise, using single-device module on a multi-GPU machine (without setting `CUDA_VISIBLE_DEVICES`) will by default lead to crash?

Reply:
What if we default to `torch.cuda.current_device()`? Seems like this would be a good default for single-device DDP module. Although, this might break multi-device DDP modules.

We should be careful when setting the default here that it doesn't break existing users for single and multi device.

Reply:
Oh, forgot the change this line. I think it will be better to use `None` as the default value, which will be consistent with multi-module mode.

The drawback of using `torch.cuda.current_device()` is the requirement of calling `torch.cuda.set_device()` in advance.